design/src/rfcs/rfc0017_customizable_client_operations.md

```diff
@@ -0,0 +1,185 @@
+RFC: Customizable Client Operations
+===================================
+
+> Status: RFC
+
+For a summarized list of proposed changes, see the [Changes Checklist](#changes-checklist) section.
+
+SDK customers occasionally need to add additional HTTP headers to requests, and currently,
+the SDK has no easy way to accomplish this. At time of writing, the lower level Smithy
+client has to be used to create an operation, and then the HTTP request augmented on
+that operation type. For example:
+
+```rust
+let input = SomeOperation::builder().some_value(5).build()?;
```

Comment:
```suggestion
let input = SomeOperationInput::builder().some_value(5).build()?;
```

I think it's easy for people to get confused about operations vs inputs and so we should clarify that this is an input for an operation and not the operation itself.

Reply:
Yeah, I intended this to be the input. Good catch.